CMakeLists.txt

```diff
@@ -33,10 +33,17 @@ if(ENABLE_TESTING)
   enable_testing()
 endif()
 
-# suppress warnings
+# build flags
 
 add_compile_options(
   $<$<AND:$<STREQUAL:$<COMPILE_LANGUAGE>,CXX>,$<CXX_COMPILER_ID:AppleClang>>:-Wno-deprecated-declarations>
+  $<$<AND:$<STREQUAL:$<COMPILE_LANGUAGE>,CXX>,$<CXX_COMPILER_ID:AppleClang>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.3>>:-Werror>
```

Comment:
Could we guard that with a CMake option that will be enabled for CI? It is usually good practice to not use `-Werror` on the users compilers by default. Otherwise distributions will have to work-around it to not make their builds explode with newer compilers or uncommon platforms/architectures.

Reply:
yes, we could do it.

Reply:
done